marge/app.py

```diff
@@ -91,15 +91,19 @@ def _secret_auth_token_and_ssh_key(options):
     else:
         auth_token = options.auth_token_file.readline()
 
-    with tempfile.NamedTemporaryFile(mode='w', prefix='ssh-key-') as env_ssh_key_file:
-        ssh_key_file = options.ssh_key_file
-        if not ssh_key_file:
-            ssh_key = os.getenv('MARGE_SSH_KEY')
-            assert ssh_key, "You need to pass --ssh-key-file or set envvar MARGE_SSH_KEY"
-            env_ssh_key_file.write(ssh_key + '\n')
-            env_ssh_key_file.flush()
-            ssh_key_file = env_ssh_key_file.name
+    ssh_key_file = options.ssh_key_file
+    if ssh_key_file:
```

Comment:
I don't like the use of the temporary file being conditional. Why not always create a temporary file?